linkml/transformers/rollup_transformer.py

```diff
@@ -0,0 +1,147 @@
+"""
+
+import copy
+from dataclasses import dataclass
+from typing import Optional
+
+from linkml_runtime.linkml_model.meta import (
+    ClassDefinitionName,
+    SchemaDefinition,
+    SlotDefinition,
+)
+from linkml_runtime.utils.schemaview import SchemaView
+
+from linkml.transformers.model_transformer import ModelTransformer
+
+
+@dataclass
+class FlattenTransformerConfiguration:
+    """
+    Configuration options for the FlattenTransformer.
+
+    Attributes:
+        preserve_class_designator: If True, preserves the original class name in the specified
+            class designator slot (e.g., 'category').
+        class_designator_slot: The name of the slot that designates the original class (e.g., 'category').
+        include_all_classes: If True, keeps all classes in the schema; otherwise, only the target class
+            and non-descendants are kept.
+        include_mixins: If True, includes properties from mixins in the flattened class.
+    """
+
+    preserve_class_designator: bool = True
+    class_designator_slot: Optional[str] = "category"
+    include_all_classes: bool = False
+    include_mixins: bool = True
+
+
+class RollupTransformer(ModelTransformer):
+    """
+    Transforms a LinkML schema by flattening an inheritance hierarchy into a single class.
+
+    This transformer takes a schema and a target class name, and produces a new schema where
+    all descendant classes of the target class are removed and their slots are rolled up into the
+    target class.
+    """
+
+    def __init__(
+        self,
+        target_class: str,
+        config: Optional[FlattenTransformerConfiguration] = None,
+    ):
+        super().__init__()
+        self.target_class = target_class
+        self.config = config or FlattenTransformerConfiguration()
+        self._descendant_classes: set[str] = set()
+        self._class_to_slot_map: dict[str, set[str]] = {}
+        self._collect_slots_for_class: dict[str, list[SlotDefinition]] = {}
+
+    def transform(self, tgt_schema_name: str = None, top_class: ClassDefinitionName = None) -> SchemaDefinition:
+        """
+        Transform the schema by flattening the inheritance hierarchy.
+
+        Args:
+            tgt_schema_name: Optional target schema name (not used in this transformer)
+            top_class: Optional top class name (not used in this transformer)
+
+        Returns:
+            A transformed SchemaDefinition with the target class having all slots from its descendants.
+        """
+        if not self.schemaview:
+            raise ValueError("SchemaView not set. Ensure schema is set by calling set_schema().")
+
+        schema_copy = copy.deepcopy(self.source_schema)
+        view = SchemaView(schema_copy)
+
+        self._identify_descendants(view)
+        self._collect_slots(view)
+        return self._create_flattened_schema(schema_copy, view)
+
+    def _identify_descendants(self, view: SchemaView) -> None:
```

Comment:
This can be done with class_descendants in SchemaView (I think? unless something extra is happening here)

Reply:
Sure! Good catch, I was not aware this exists.

Reply:
This looks like it meets @kevinschaper request.